### ISSUE TITLE
Watson Designer: UX Stories

### ISSUE THREAD

**Issue by joshkimmell** [punchcard-cms/punchcard#245]
{{long description}}

```gherkin
Feature: {{feature}}
  As a {{persona}}
  I want {{need}}
  So that {{rationale}}

  Scenario: {{scenario}}
    Given {{thing}}
     When {{action}}
     Then {{result}}
```

---

**{{feature}}**
- [ ] {{scenario}}

## General - Nick
* - [x] Wayfinding
  * Let user know they are logged in (display name in header) [#217](https://github.com/punchcard-cms/punchcard/issues/217)
  * Make sure user knows what page/section they are on (breadcrumbs, affordance on main nav) [#95](https://github.com/punchcard-cms/punchcard/issues/95) + [#219](https://github.com/punchcard-cms/punchcard/issues/219)
* - [x] Messaging (alerts around content related to user) [#171](https://github.com/punchcard-cms/punchcard/issues/171) + [#172](https://github.com/punchcard-cms/punchcard/issues/172)
* - [x] First time user experience (what do I do?) [#237](https://github.com/punchcard-cms/punchcard/issues/237)
* - [x] Display user name of content creator/editor on all content [#234](https://github.com/punchcard-cms/punchcard/issues/234)
* - [x] Date format could be more human readable (Thu Jul 28 2016 20:12) [#239](https://github.com/punchcard-cms/punchcard/issues/239)
* - [x] Should we be more specific for the "Date" column? Maybe "Last edited" or something. [#239](https://github.com/punchcard-cms/punchcard/issues/239)

## Login - Nick
* - [x] Forgot password [#148](https://github.com/punchcard-cms/punchcard/issues/148)
* - [x] Receive an email when I am added to Punchcard [#149](https://github.com/punchcard-cms/punchcard/issues/149) + [#167](https://github.com/punchcard-cms/punchcard/issues/167)
* - [x] Request access [#206](https://github.com/punchcard-cms/punchcard/issues/206) + [#207](https://github.com/punchcard-cms/punchcard/issues/207)
* - [x] Failed login (no error message) [#225](https://github.com/punchcard-cms/punchcard/issues/225)

## Create content - Nick
* - [ ] Content types list (provide more context and allows users to perform their tasks from this screen. what could the user do here, without having to drill down another level)
* - [ ] Content form
  * What should the hierarchy be on this page? Currently the main form and the sunrise/sunset portion share equal weight. Is that right? What happens with more complex content types.
  * What fields are required? Need some visual affordance.
  * Discuss + explore: Save vs Save and continue (autosave) vs Ready to publish (publish could happen here as well as the revisions list view)
* - [ ] What happens to content once it goes into the approval workflow? Does it lock? Can you pull that content back?
* - [x] Where can you see a list of all the content you are currently working on? (dashboard?) [#228](https://github.com/punchcard-cms/punchcard/issues/228)
* - [x] Do we need a place to comment for each create/edit of content? [#232](https://github.com/punchcard-cms/punchcard/issues/232)
* - [x] Add character limits on form elements [#193](https://github.com/punchcard-cms/punchcard/issues/193)

## Edit content - Nick
* - [ ] Content type history
  * What is the order here? Hard to understand what is the "correct" version.
  * Need a way to see the "status" of a piece of content (draft, submitted for approval, approved)
* - [x] Need a warning when you leave a page with content you entered that was not “saved” [#241](https://github.com/punchcard-cms/punchcard/issues/241)

## Revisions - Miranda
* - [x] Discuss the locking content on content [#170] (https://github.com/punchcard-cms/punchcard/issues/170)
* - [x]  How can the revision history / conversation not get lost when you make a new revision? [#223] (https://github.com/punchcard-cms/punchcard/issues/223)
* - [x] What happens when something is approved, does that content and all the revisions for it change in someway? Can you still edit it? Does that become a new "revision" or a new piece of content? [#170](https://github.com/punchcard-cms/punchcard/issues/170) - needs more stories
* - [x] If you create a revision but make no edits - should the save and publish option not be available? - seems to be related to issue [#221] (https://github.com/punchcard-cms/punchcard/issues/221) - where it addresses saving a revision, sending a revision into a workflow or canceling the creation of a revision.
* - [ ] When you create a new revision, what revision of the content does it use?
* - [x] DIFF - compare revisions and edits in content (like GitHub does) - also listed under Approval section [#224] (https://github.com/punchcard-cms/punchcard/issues/224)
* - [x] A way to recall a submission for approval [#230] (https://github.com/punchcard-cms/punchcard/issues/230)
* - [x] Saving a revision without sending into workflow [#221] (https://github.com/punchcard-cms/punchcard/issues/221)

## Approval - Miranda
* - [x] The approve screen needs some work. Discuss having a way to display the "diff" between what was already their and what was changed. Content creator diff: [#244] (https://github.com/punchcard-cms/punchcard/issues/224) for diff; Editor/Approver UI + Diff [#233] (https://github.com/punchcard-cms/punchcard/issues/233)
* - [x] Approval actions: Approve / Reject - is that the only actions we need? "Request revisions"? "Comment"? [#233] (https://github.com/punchcard-cms/punchcard/issues/233) - Connected this to the approval screen/UI issue #233.
* - [x] As an approver or content creator, how do you know if there is a change to your content? (dashboard? notifications?) Content creator: [#171] (https://github.com/punchcard-cms/punchcard/issues/171) Approver/editor: [#172] (https://github.com/punchcard-cms/punchcard/issues/172)
* - [x] If a piece of content is waiting for approval, can you notify the "approver" again? (send reminder) [#235] (https://github.com/punchcard-cms/punchcard/issues/235)
* - [x] Need an approval log or area for approvers to approve and review items - connected to [#233] (https://github.com/punchcard-cms/punchcard/issues/233)
* - [x] Need a way to keep track of comments across revisions - see issue created for Content Creator [#223] (https://github.com/punchcard-cms/punchcard/issues/223)

## Users - Miranda
* - [x] Send email to new users that are created - Core Developer Issue [#149] (https://github.com/punchcard-cms/punchcard/issues/149)
* - [ ] Users shouldn't be able to edit themselves [#163](https://github.com/punchcard-cms/punchcard/issues/163) - might need to update this story with this issue
* - [x] How are the roles defined & created? - May need to add to this user admin issue [#163] (https://github.com/punchcard-cms/punchcard/issues/163)
* - [ ] How does a user know what role is best for that user? Can we display what each role does?
* - [x] Username of item creator should show up in content list so everyone can see who created what item - linking to existing story about displaying content creator's email - could potentially be one and the same, or added to that issue [#234] (https://github.com/punchcard-cms/punchcard/issues/234)
* - [x] Edit + Delete current don't work (bug) [#166](https://github.com/punchcard-cms/punchcard/issues/166)
* - [ ] Will users from different teams exist in different orgs/groups/depts within the CMS?
* - [ ] What types of content will users from different teams be able to see? Will all users be able to see everything?
* - [ ] Allow users to change their own passwords [#165](https://github.com/punchcard-cms/punchcard/issues/165) - need a new story, but #165 is similar.

**Comment by ineedsubstance** [comment]
@joshkimmell I think you added this, but I'm not sure what it means.

![screen shot 2016-08-16 at 11 11 10 am](https://cloud.githubusercontent.com/assets/3981381/17706578/32d1a1d0-63a2-11e6-8acf-274e662bee26.png)

**Comment by joshkimmell** [comment]
It means we need the ability to "delete" a piece of content, from the user's perspective.

**Comment by ineedsubstance** [comment]
Wouldn't `sunset` accomplish this?

**Comment by joshkimmell** [comment]
no... it will still remain in the list of content. For instance, Conversation and Watson Conversation API... which one is the right one? Allen wrote one and I wrote the other. I guess a temporary solution would be to title it something like "Conversation - DO NOT USE", but the list will get long and cluttered. The story doesn't need to say "Archive".

**Comment by ineedsubstance** [comment]
ok, so you are saying `delete` it for the content creator. got it. thanks.

**Comment by ineedsubstance** [comment]
I will flush out that story. Seems like something we would only want certain users to be able to do.
